src/lib-components/forms/RadioCards.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+import Uniques from "@/helpers/Uniques"
+import {
+  RadioGroup,
+  RadioGroupDescription,
+  RadioGroupLabel,
+  RadioGroupOption,
+} from "@headlessui/vue"
+import { CheckCircleIcon } from "@heroicons/vue/solid"
+import { computed, ref, useAttrs } from "vue"
+import InputLabel from "./InputLabel.vue"
+import InputHelp from "./InputHelp.vue"
+import FieldsetLegend from "./FieldsetLegend.vue"
+
+/*
+ * NOTE (spk) headless UI introduced a "name" prop that includes a hidden field
+ * to use the modelValue inside of forms.  It does not however resolve the issue of
+ * supporting HTML5 form validation, so we'll add our own hidden radio buttons to support both.
+ *
+ * The headless technique does include supporting complex modelValues such as objects, which we may
+ * need in the future.  We can revist required validation at that time using a singular hidden checkbox.
+ */
+
+type ModelValue = string | number
+
+type RadioCard = {
+  disabled?: boolean
+  help?: string
+  label: string
+  sublabel?: string
+  value: ModelValue
+}
+
+const props = withDefaults(
+  defineProps<{
+    columns?: 2 | 3
+    help?: string
+    legend?: string
+    modelValue?: ModelValue
+    options: RadioCard[]
+  }>(),
+  {
+    columns: undefined,
+    help: "",
+    legend: "",
+    modelValue: undefined,
+  }
+)
+
+const emit = defineEmits<{
+  (e: "update:modelValue", modelValue: ModelValue): void
+}>()
+
+const attrs = useAttrs()
+const uuid = (attrs.id as string) || Uniques.CreateIdAttribute()
```

Comment:
any reason to always use: Uniques.CreateIdAttribute()? IME, it's not uncommon to copy pasta the same id by accident in a form

Reply:
Good point, we had some cases where it made sense on basic form components, but we never merge attributes here, so it's not as useful.  Let's use `Uniques` until we find a reason to allow a custom ID attribute.